docker_mcp/core/migration/volume_parser.py

```
@@ -0,0 +1,262 @@
+            volumes_info["bind_mounts"] = list(set(volumes_info["bind_mounts"]))
+            
+            self.logger.info(
+                "Parsed compose volumes",
+                named_volumes=len(volumes_info["named_volumes"]),
+                bind_mounts=len(volumes_info["bind_mounts"]),
+            )
+            
+            return volumes_info
+            
+        except yaml.YAMLError as e:
+            raise VolumeParsingError(f"Failed to parse compose file: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)